experimental/packages/opentelemetry-sdk-metrics-base/README.md

```diff
@@ -21,6 +21,57 @@ Please see the [version `0.27.0` README](https://github.com/open-telemetry/opent
 
 TODO: Add usage information for updated SDK
 
+## Installation of the Latest experimental version
+
+```bash
+npm install --save @opentelemetry/sdk-metrics-base
+```
+
+## Usage of the Latest experimental version
+
+The basic setup of the SDK can be seen as followings:
+
+```js
+const opentelemetry = require('@opentelemetry/api-metrics');
+const { MeterProvider } = require('@opentelemetry/sdk-metrics-base');
+
+// To create an instrument, you first need to initialize the Meter provider.
+// NOTE: The default OpenTelemetry meter provider does not record any metric instruments.
+//       Registering a working meter provider allows the API methods to record instruments.
+opentelemetry.setGlobalMeterProvider(new MeterProvider());
+
+// To record a metric event, we used the global singleton meter to create an instrument.
+const counter = opentelemetry.getMeter('default').createCounter('foo');
+
+// record a metric event.
+counter.add(1, { attributeKey: 'attribute-value' });
+```
+
+In conditions, we may need to setup an async instrument to observe costly events:
+
+```js
+// Creating an async instrument, similar to synchronous instruments
+const observableCounter = opentelemetry.getMeter('default')
+  .createObservableCounter('observable-counter');
+
+// Register a single-instrument callback to the async instrument.
+observableCounter.addCallback(async (observableResult) => {
+  // ... do async stuff
+  observableResult.observe(1, { attributeKey: 'attribute-value' });
+});
+
+// Register a multi-instrument callback and associate it with a set of async instruments.
+opentelemetry.getMeter('default')
```

Comment:
So you have to keep a reference to your counter somehow? Maybe we should provide the list (in the same order) on the batchObservableResult object? The experience might even be improved if you could name them and provide it as a map instead of a list?

I am imagining a case like this:

```typescript
// Creating an async instrument, similar to synchronous instruments
const observableCounter = opentelemetry.getMeter('default')
  .createObservableCounter('observable-counter');

const observableCounter2 = opentelemetry.getMeter('default')
  .createObservableCounter('observable-counter-2');

// Register a multi-instrument callback and associate it with a set of async instruments.
opentelemetry.getMeter('default')
  .addBatchObservableCallback(batchObservableCallback, { counter1: observableCounter, counter2: observableCounter2 });

async function batchObservableCallback(batchObservableResult) {
  const result = await doSomethingExpensive();

  batchObservableResult.observe(batchObservableResult.instruments.counter1, result.result1, { attributeKey: 'attribute-value-1' });
  batchObservableResult.observe(batchObservableResult.instruments.counter2, result.result2, { attributeKey: 'attribute-value-2' });
}
```

i'm not sure how much additional work this would be to implement so just take this as a suggestion. I'm still going to leave my approval

Reply:
This can create another layer of concepts: the instruments (with another name) in the `batchObservableResult.instruments`, over the original observable instrument objects. That opens another question to the user: can I observe with the original observable instruments, what's the difference between them? I'd find this creates more entities and depth to the API, and personally find it harder to use correctly.

I can think of the benefit here is that we can create the batch observable callback function object without any reference to the outer context, like an instrument class with static functions. However, encapsulating references to outer scope variables or binding the function to an object which holds the references should be relatively easy in JavaScript (and it's the common practice in JS). I would say it can be cleaner and more flexible without another layer of abstraction.

Reply:
ok